File: src/snippets/bot-protection/advanced-signals/python-fastapi/BlockSignals.py
import os

from arcjet import Mode, arcjet, detect_bot
from fastapi import FastAPI, Request
from fastapi.responses import JSONResponse

app = FastAPI()

aj = arcjet(
    key=os.getenv("ARCJET_KEY"),  # Get your site key from https://app.arcjet.com
    rules=[
        detect_bot(
            mode=Mode.LIVE,
            allow=[],  # deny all bots by default, including signals failures
        ),
    ],
)


@app.post("/contact")
async def contact(request: Request):
    decision = await aj.protect(request)

    if decision.is_denied():
        return JSONResponse({"error": "Forbidden"}, status_code=403)

    return {"message": "success"}

File: src/snippets/bot-protection/advanced-signals/nodejs/BlockSignals.js
import arcjet, { detectBot } from "@arcjet/node";
import http from "node:http";

const aj = arcjet({
  key: process.env.ARCJET_KEY,
  rules: [
    detectBot({
      mode: "LIVE",
      allow: [], // deny all bots by default, including signals failures
    }),
  ],
});

const server = http.createServer(async (req, res) => {
  const decision = await aj.protect(req);

  if (decision.isDenied()) {
    res.writeHead(403, { "Content-Type": "application/json" });
    res.end(JSON.stringify({ error: "Forbidden" }));
    return;
  }

  res.writeHead(200, { "Content-Type": "application/json" });
  res.end(JSON.stringify({ message: "success" }));
});

server.listen(3000);
